src/BaseWrapper.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Thanks to ultrasecr.eth
+pragma solidity ^0.8.0;
+
+import { IERC3156PPFlashLender } from "lib/erc3156pp/src/interfaces/IERC3156PPFlashLender.sol";
+import { IERC20 } from "lib/erc3156pp/src/interfaces/IERC20.sol";
+
+import { TransferHelper } from "./utils/TransferHelper.sol";
+
+abstract contract BaseWrapper is IERC3156PPFlashLender {
+    using TransferHelper for IERC20;
+
+    struct Data {
+        address loanReceiver;
+        address initiator;
+        function(address, address, IERC20, uint256, uint256, bytes memory) external returns (bytes memory) callback;
+        bytes initiatorData;
+    }
+
+    bytes internal _callbackResult;
+
+    /// @inheritdoc IERC3156PPFlashLender
```

Comment:
We should @inheritdoc from the ERC7399 interface when possible, I think.

Reply:
Where does that interface lives? I can't see it in the deps